rosbag2_cpp/test/rosbag2_cpp/test_action_utils.cpp

```diff
@@ -0,0 +1,149 @@
+    {{"/abc/_action/cancel_goal", "action_msgs/srv/CancelGoal_Event"}, false},
+    {{"/abc/_action/cancel_goal/_service_event", "action_msgs/CancelGoal_Event"}, false},
+    {{"/abc/_action/cancel_goal/service_event", "action_msgs/srv/CancelGoal"}, false},
+    {{"/abc/_action/status", "action_msgs/msg/GoalStatusArray"}, true},
+    {{"/abc/action/status", "action_msgs/msg/GoalStatusArray"}, false},
+    {{"/abc/_action/status", "action_msgs/GoalStatusArray"}, false},
+    {{"/abc/_action/status", "action_msgs/msg/GoalStatus"}, false}
+  };
+
+  for (const auto & test_data : all_test_data) {
+    EXPECT_TRUE(
+      rosbag2_cpp::is_topic_related_to_action(
+        std::get<0>(test_data.first), std::get<1>(test_data.first)) == test_data.second);
+  }
+}
+
+TEST_F(ActionUtilsTest, check_action_topic_name_to_action_name)
+{
+  std::vector<std::pair<std::string, std::string>> all_test_data =
+  {
+    {"/abc/_action/feedback", "/abc"},
+    {"/abc/_action/get_result/_service_event", "/abc"},
+    {"/abc/_action/send_goal/_service_event", "/abc"},
+    {"/abc/_action/cancel_goal/_service_event", "/abc"},
+    {"/abc/_action/status", "/abc"},
+    {"/_action/status", ""},
+    {"/abc/action/status", ""}
+  };
+
+  for (const auto & test_data : all_test_data) {
+    EXPECT_TRUE(
+      rosbag2_cpp::action_topic_name_to_action_name(test_data.first) == test_data.second);
+  }
+}
+
+TEST_F(ActionUtilsTest, check_action_topic_type_to_action_type)
+{
+  std::vector<std::pair<std::string, std::string>> all_test_data =
+  {
+    {"package/action/xyz_FeedbackMessage", "package/action/xyz"},
+    {"package/action/xyz_GetResult_Event", "package/action/xyz"},
+    {"package/action/xyz_SendGoal_Event", "package/action/xyz"},
+    {"action_msgs/srv/CancelGoal_Event", ""},
+    {"action_msgs/msg/GoalStatusArray", ""}
+  };
+
+  for (const auto & test_data : all_test_data) {
+    EXPECT_EQ(
+      rosbag2_cpp::action_topic_type_to_action_type(test_data.first),
+      test_data.second
+    );
+  }
+}
+
+TEST_F(ActionUtilsTest, check_get_action_topic_type_from_topic_name)
+{
+  std::vector<std::pair<std::string, rosbag2_cpp::TopicsInAction>> all_test_data =
+  {
+    {"/abc/_action/feedback", rosbag2_cpp::TopicsInAction::Feedback},
+    {"/abc/_action/get_result/_service_event", rosbag2_cpp::TopicsInAction::GetResultEvent},
+    {"/abc/_action/send_goal/_service_event", rosbag2_cpp::TopicsInAction::SendGoalEvent},
+    {"/abc/_action/cancel_goal/_service_event", rosbag2_cpp::TopicsInAction::CancelGoalEvent},
+    {"/abc/_action/status", rosbag2_cpp::TopicsInAction::Status},
+    {"/_action/status", rosbag2_cpp::TopicsInAction::Unknown},
+    {"/abc/action/status", rosbag2_cpp::TopicsInAction::Unknown}
+  };
+
+  for (const auto & test_data : all_test_data) {
+    EXPECT_EQ(
+      rosbag2_cpp::get_action_topic_type_from_topic_name(test_data.first),
+      test_data.second
+    );
```

Comment:
Nitpick: This statement could be in one line instead of 4
```suggestion
    EXPECT_EQ(rosbag2_cpp::get_action_interface_type(test_data.first), test_data.second);
```